swimster-ui/src/hooks/useLoginForm.js

```diff
@@ -0,0 +1,47 @@
+import React, { useState, useEffect } from 'react'
+import { useNavigate } from 'react-router-dom'
+import  { useForm } from 'react-hook-form'
+import apiClient from '../services/apiClient'
+import { useAuthContext } from '../contexts/auth'
+
+export const useLoginForm = () => {
+    const { user, setUser } = useAuthContext()
+    const navigate = useNavigate()
+    const { register, handleSubmit } = useForm()
+    const [isProcessing, setIsProcessing] = useState(false)
+    const [error, setError] = useState({})
+    const [form, setForm] = useState({})
+
+    useEffect(() => {
+        // if user is logged in, redirect them to Home
```

Comment:
let's also add comments across the whole codebase for consistency

Reply:
got it